src/engine/objects/mod.rs

```diff
@@ -210,11 +210,20 @@ where
     /// }
     /// ```
     pub fn deser<T: serde::de::DeserializeOwned>(&self) -> Result<T, Error> {
+        /*
         let m = Value::Map(self.fields().clone());
         let v = serde_json::Value::try_from(m)?;
         let t: T = serde_json::from_value(v)
             .map_err(|e| Error::JsonDeserializationFailed { source: e })?;
         Ok(t)
+        */
+        let mut fields = self.fields().clone();
+        fields.insert("__label".to_string(), Value::String(self.concrete_typename.clone()));
```

Comment:
Is the idea that this corresponds to the GraphQL `__typename` attribute?  Is the name `__label` a Warpgrapher-specific convention?  Is idea is that someone writing a deserializer from a `serde_json::Value` could do something like key off of label to decide a specific `enum` variant to use, or something like that?

Reply:
The last one (it helps deserialize enums).